src/Illuminate/Cookie/Middleware/EncryptCookies.php

```diff
@@ -214,7 +216,15 @@ protected function duplicate(Cookie $cookie, $value)
      */
     public function isDisabled($name)
     {
-        return in_array($name, array_merge($this->except, static::$neverEncrypt));
+        $exclude = Collection::make($this->except)
+            ->merge(static::$neverEncrypt)
+            ->filter(fn ($cookie) => Str::contains($cookie,'*'));
+
+        return $exclude->isEmpty()
+            ? in_array($name, array_merge($this->except, static::$neverEncrypt))
+            : $exclude->filter(fn ($cookie) => Str::startsWith(
+                $name, Str::replace('*', '', $cookie)
+            ))->isNotEmpty();
```

Comment:
This reduces redundant code:
```suggestion
        $except = Collection::make($this->except)->merge(static::$neverEncrypt);
        $exclude = $except->filter(fn ($cookie) => Str::contains($cookie,'*'));

        return $exclude->isEmpty()
            ? $except->contains($name)
            : $exclude->filter(fn ($cookie) => Str::startsWith(
                $name, Str::replace('*', '', $cookie)
            ))->isNotEmpty();
```

Reply:
This might even be rewritten to (especially, together with the other suggestion):
```suggestion
        $except = Collection::make($this->except)->merge(static::$neverEncrypt);

        return $except->some(fn ($cookie) => Str::contains($cookie,'*'))
            ? $except->contains($name)
            : $except->filter(fn ($cookie) => Str::startsWith(
                $name, Str::replace('*', '', $cookie)
            ))->isNotEmpty();
```

Reply:
@shaedrich Thanks for your input, it looks better now.

Reply:
@selcukcukur You're welcome 👍🏻

Reply:
Oh, wait, sorry: The `some` has to be inverted to `doesntContain` 🤦🏻‍♂️

Reply:
> Oh, wait, sorry: The `some` has to be inverted to `doesntContain` 🤦🏻‍♂️

Yes, I noticed it while doing the tests and now I'm interested.